src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/spec/response/confirmation/CounterClaimConfirmationText.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.civil.handler.callback.user.spec.response.confirmation;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.civil.enums.RespondentResponseTypeSpec;
+import uk.gov.hmcts.reform.civil.enums.YesOrNo;
+import uk.gov.hmcts.reform.civil.handler.callback.user.spec.RespondToClaimConfirmationTextSpecGenerator;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+
+import java.util.Optional;
+
+@Component
+public class CounterClaimConfirmationText implements RespondToClaimConfirmationTextSpecGenerator {
+
+    @Override
+    public Optional<String> generateTextFor(CaseData caseData) {
+        if (!isCounterClaim(caseData)) {
+            return Optional.empty();
+        }
+
+        StringBuilder sb = new StringBuilder();
```

Comment:
This text is always the same, could be a class' constant instead of being calculated in a StringBuilder